### PR TITLE
Update Sphinx and docs.in / docs.txt dependency trees

### PR DESCRIPTION
**Pull request by ScriptAutomate**
### What does this PR do?

Updates Sphinx to the latest version, and updates the `requirements/*/docs.txt` from `pre-commit` based on the updated dependency trees.

### What issues does this PR fix or reference?

This issue was discovered while resolving:
- #59798

It is also a prereq as part of work with:
- #59451

### Previous Behavior

Sphinx and the dependency tree for Sphinx were running out-dated packages, some of which had versions in conflict with other dependency trees in this repository.

### New Behavior

Updating Sphinx resolves problems discovered with `safety` (at least, the problems found the `docs.txt` req trees), and also removes the conflicting pinned versions of packages between different dependency trees in the repo.

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [x] Docs
- [ ] Changelog - https://docs.saltstack.com/en/master/topics/development/changelog.html
- [ ] Tests written/updated

### Commits signed with GPG?
Yes

### CODE REVIEW

**Review by s0undt3ch — DISMISSED** [review]

**Comment by ScriptAutomate** [comment]
The following error happens when attempting to use `nox -e 'docs-html(compress=False, clean=True)'`:

```
...
...
writing output... [100%] topics/yaml/index


Warning, treated as error:
<rst_prolog>:1:unknown document: /topics/releases/latest_release
make: *** [Makefile:56: html] Error 2
nox > Command make html SPHINXOPTS=-W failed with exit code 2
nox > Session docs-html(compress=False, clean=True) failed.
```

Troubleshooting to see whether upgrading Sphinx led to this break or not.

**Comment by ScriptAutomate** [comment]
On a clean salt repo, we can successfully run:
```
nox -e 'docs-html(compress=False, clean=True)'
```
I was going to see about updating Sphinx in this PR, but I ran into a problem. It looks like it breaks things, and the pipelines may have been working because Sphinx wasn't yet flagging something that is rightly a problem:

https://github.com/saltstack/salt/blob/736d39cf425000f1981ded0546a9bb348496ca1f/doc/conf.py#L49-L51
- Example error after Sphinx update: https://github.com/saltstack/salt/runs/2126496854?check_suite_focus=true

This isn't a problem in `builddocs` because `build_env.sh` exports a `LATEST_RELEASE` var before it ever gets to running Sphinx, but this means trying to build docs when developing in the `salt` repo, or when working with GitHub Actions will break with Sphinx unless I allow Sphinx to run with warnings (it produces over 1700 warnings due to this one problem, when working with the latest version of Sphinx). I'm not sure what the best path is. A user could just set it in the `conf.py` as the default fallback, to be an existing version, or they can be in an environment with the `LATEST_RELEASE` env var set to an appropriate value. But that adds extra steps to anyone contributing, and is purely adhoc. If I set a default version to always work, then we would have to make sure we never accidentally remove or mess with `LATEST_RELEASE` in the builddocs repo, as it would fallback to believing that the latest is `3002.5` (or whatever number we choose), even when it isn't really the latest version, and it would do so without throwing warnings in output.

Some paths I can imagine:
- Set the default value of `LATEST_RELEASE` in the `conf.py` file. The env var for `LATEST_RELEASE` will override
- Pull the `LATEST_RELEASE` value from `builddocs` somehow: https://gitlab.com/saltstack/open/docs/builddocs/-/blob/master/build_env.sh#L8
  - Though, this would mean `nox` is reaching out to the internet to retrieve values. Not a good idea in my opinion, and is now fragile based on changes to `builddocs`
- Allow ignoring of warnings with Sphinx during `nox` builds, so that it only fails when there is an error: least ideal, as we now will miss ALL warnings
- I don't want to be stuck updating the `Makefile` or the `conf.py` file in this repo whenever there is a newer `LATEST_RELEASE`, but I don't know how to automate this because I don't know how to retrieve the latest version dynamically from within this repo anywhere. Is there a way to make use of `saltstack/salt/salt/version.py` to retrieve this? If so, would it be a bad idea?

**I decided to just set the default value to `3002.5`.** The pipeline succeeds, and it can be overridden with an env var, such as what is used by `builddocs`.

**Comment by s0undt3ch** [comment]
@ScriptAutomate I have update the PR to pull in the latest release version from the code checkout, ie, no need to hardcode and require updating versions going forward.

File: doc/conf.py [hunk 1]
```diff
@@ -42,19 +42,20 @@
...
-latest_release = os.environ.get(
-    "LATEST_RELEASE", "latest_release"
-)  # latest release (2019.2.3)
+# This is the default branch on GitHub for the Salt project
+repo_primary_branch = "master"
+latest_release = (
+    os.environ.get("LATEST_RELEASE")
+    or
+    # The next unreleased version
+    str(salt.version.__saltstack_version__.major)
```
Comment (bryceml) [inline doc/conf.py:51]: What would this end up being at various times? is the fact that we have a recent RC change this?

Comment (ScriptAutomate) [reply doc/conf.py:51]: This is mostly for local builds. Right now, for example, it returns `3003`. When we build, and supply `latest_release` as an env var, it takes that value instead. Before now, this was left empty and defaulting to `'latest_release'` as a textual placeholder, which Sphinx previously didn't care about, but newer versions of Sphinx throw a warning when they see that there is no release file:

```
Warning, treated as error:
<rst_prolog>:1:unknown document: /topics/releases/latest_release
make: *** [Makefile:56: html] Error 2
```

**Review by bryceml — APPROVED** [review]

**Review by s0undt3ch — APPROVED** [review]

**Review by Ch3LL — APPROVED** [review]

**Pull request merged**